### ISSUE TITLE
Unit Tests for GPT API

### ISSUE THREAD

**Issue by uc-mcreyntm** [brendanpayne/protifyapp#31]
### Story Description:
Create Unit tests for GPT api call

#### Acceptance Criteria:

1. Unit Tests for API input passing

2.  Unit Tests for API output passing

#### Additional Information:

Unit tests need to cover all functionality of the AI, such as not scheduling outdoor events while it's raining, and making sensical times for your events according to efficient routing of driving.

#### Tasks:

Create unit tests that cover core AI functionality

#### Definition of Done:

All created unit tests are compresensive and passing

**Comment by uc-mcreyntm** [comment]
Created a basic unit test for essential AI functionality. This test will sync with the events in the database on February 20th, 2024 and use the events on that day for testing. 

Tests for succesful API response (doesn't matter what  the output is)
Tries to parse AI output into a structured class
Tests to see if the outputted schedule is different from the original

Started working on more comprehensive testing for when events are outdoor and it's supposed to rain for that day. All of the tests run assuming its raining from midnight to 4pm. It will pull the events from February 24th, 2024 and will ensure that at least one of the events are labled as outdoor. I am working on a function that make sure the response is valid in format and is not the same as the original, as well as making sure that none of the outdoor events are scheduled during raining hours.

**Comment by uc-mcreyntm** [comment]
Fixed parsing errors in the strict testing unit tests so they would run. Tweaked prompt until I was able to get unit tests to pass 😎 . Here is the current passing criteria. Keep in mind this unit test is only for when it is supposed to rain that day and the user has 1 or more outdoor events. I currently have two separate functions that have to be called independantly of each other to determine this. Down the line, I will add a parameter to the class in the form of a bool to determine which prompt it should use.

1. The old schedule (what the user put in) is different from the new schedule (what the ai sugguests)
2.   The old or new schedule isn't empty (This is mainly to make sure that the AI output a parseable json string and it was able to be parsed into the data class)
3.  All of the events given to the AI are still in the response (None of the events were "deleted")
4.  If there is one outdoor event, it is not scheduled during a time that is raining. If there is more than one outdoor event, at least one of them is scheduled during a time that is non raining
5. The duration of the events is unchanged (For example, if the user makes a 3 hour long event, it remains three hours long)
